src/Experiment.tsx

```diff
@@ -14,167 +14,67 @@
-      };
-    }
+const Experiment: React.FunctionComponent<ExperimentProps> = props => {
+  const { experiment, autoUpdate, timeout, overrideUserId, overrideAttributes, children } = props;
+  const [variation, clientReady, didTimeout] = useExperiment(
+    experiment,
+    { timeout, autoUpdate },
+    { overrideUserId, overrideAttributes }
+  );
+
+  if (!clientReady && !didTimeout) {
+    // Only block rendering while were waiting for the client within the allowed timeout.
+    return null;
   }
 
-  componentDidMount() {
-    const { experiment, optimizely, optimizelyReadyTimeout, isServerSide, timeout } = this.props;
-    if (!optimizely) {
-      throw new Error('optimizely prop must be supplied');
-    }
-    if (isServerSide) {
-      return;
-    }
-
-    // allow overriding of the ready timeout via the `timeout` prop passed to <Experiment />
-    const finalReadyTimeout: number | undefined = timeout !== undefined ? timeout : optimizelyReadyTimeout;
-
-    optimizely.onReady({ timeout: finalReadyTimeout }).then((res: OnReadyResult) => {
-      if (res.success) {
-        logger.info('experiment="%s" successfully rendered for user="%s"', experiment, optimizely.user.id);
-      } else {
-        logger.info(
-          'experiment="%s" could not be checked before timeout of %sms, reason="%s" ',
-          experiment,
-          timeout === undefined ? DEFAULT_ON_READY_TIMEOUT : timeout,
-          res.reason || ''
-        );
-      }
-
-      const variation = optimizely.activate(experiment);
-      this.setState({
-        canRender: true,
-        variation,
-      });
-      if (this.autoUpdate) {
-        this.setupAutoUpdateListeners();
-      }
-    });
+  if (children != null && typeof children === 'function') {
+    // Wrap the return value here in a Fragment to please the HOC's expected React.ComponentType
+    // See https://github.com/DefinitelyTyped/DefinitelyTyped/issues/18051
+    return <>{(children as ChildRenderFunction)(variation, clientReady, didTimeout)}</>;
   }
 
-  setupAutoUpdateListeners() {
-    const { optimizely, experiment } = this.props;
-    if (optimizely === null) {
-      return;
-    }
-
-    this.optimizelyNotificationId = optimizely.notificationCenter.addNotificationListener(
-      'OPTIMIZELY_CONFIG_UPDATE',
-      () => {
-        logger.info(
-          'OPTIMIZELY_CONFIG_UPDATE, re-evaluating experiment="%s" for user="%s"',
-          experiment,
-          optimizely.user.id
-        );
-        const variation = optimizely.activate(experiment);
-        this.setState({
-          variation,
-        });
-      }
-    );
-
-    this.unregisterUserListener = optimizely.onUserUpdate(() => {
-      logger.info('User update, re-evaluating experiment="%s" for user="%s"', experiment, optimizely.user.id);
-      const variation = optimizely.activate(experiment);
-      this.setState({
-        variation,
-      });
-    });
-  }
+  let match: React.ReactElement<VariationProps> | null = null;
```

Comment:
Github is being dramatic here. None of this was changed from the render method of the class component.